src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -395,16 +395,20 @@ class SolverBase : public ExecutableGroup
 
   /**
    * @brief calculate the norm of the global system residual
-   * @param rhs the system right-hand side vector
-   * @param dofManager degree-of-freedom manager associated with the linear system
+   * @param time the time at the beginning of the step
+   * @param dt the desired timestep
    * @param domain the domain partition
+   * @param dofManager degree-of-freedom manager associated with the linear system
+   * @param localRhs the system right-hand side vector
    * @return norm of the residual
    *
    * This function returns the norm of global residual vector, which is suitable for comparison with
    * a tolerance.
    */
   virtual real64
-  calculateResidualNorm( DomainPartition const & domain,
+  calculateResidualNorm( real64 const & time,
```

Comment:
In fact here, it is quite convenient to have a dummy definition of `calculateResidualNorm` for the `SurfaceGenerator` as well as for the explicit solvers, since they don't care about residual norms.